docs/implementation.md

```diff
@@ -0,0 +1,210 @@
+## Implementing Links
+
+Before implementing your own Link, be sure to read the [usage](usage.md) document.
+
+Links implement a `request` method that takes an operation and returns the results in an Observable.
+The apollo-link's `Observable` follows the ECMAScript [proposal](https://github.com/tc39/proposal-observable#api) with some extensions provided by [zen-observable](https://github.com/zenparsing/zen-observable#api).
```

Comment:
I wonder if it's worth a _further_ extension of exposing a `then` and `catch` for `Promise`-like semantics for the simple cases where a caller just wants to make a straight network request w/o any caching or subscriptions